tv_apps/tv_shared/js/card_manager.js

```diff
@@ -210,14 +197,14 @@
       var that = this;
       return this._getPipedPromise('_reloadCardList',
         function(resolve, reject) {
+          that._asyncSemaphore.v();
           // initialize card store if needed
           if (!that._cardStore) {
             that._cardStore =
               new CardStore(that._mode, that._manifestURLOfCardStore);
             that._cardStore.on('change', that._onCardStoreChange.bind(that));
           }
-          that.state = CardManager.STATES.SYNCING;
-          that._cardStore.getData('cardList').then(function(cardList) {
+          return that._cardStore.getData('cardList').then(function(cardList) {
```

Comment:
The way of nesting promise here is just too complex. I would suggest to refactor it and avoid resolve a promise in another nested promise.